addons/binding/org.openhab.binding.feed/src/main/java/org/openhab/binding/feed/handler/FeedHandler.java

```diff
@@ -0,0 +1,290 @@
+     */
+    private SyndFeed fetchFeedData(String urlString) {
+        SyndFeed feed = null;
+        try {
+            URL url = new URL(urlString);
+
+            URLConnection connection = url.openConnection();
+
+            BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+
+            SyndFeedInput input = new SyndFeedInput();
+            feed = input.build(in);
+            in.close();
+
+            if (this.thing.getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (MalformedURLException e) {
+            logger.warn("Url '{}' is not valid: {}", urlString, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.getMessage());
+            return null;
+        } catch (IOException e) {
+            logger.warn("Error accessing feed: " + urlString, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.getMessage());
+            return null;
+        } catch (IllegalArgumentException e) {
+            logger.warn("Feed URL is null", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.getMessage());
+            return null;
+        } catch (FeedException e) {
+            logger.warn("Feed content is not valid: " + urlString, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.getMessage());
+            return null;
+        }
+
+        return feed;
+
+    }
+
+    /**
+     * Gets the title, description and the published date of the most recent entry.
+     */
+    private String getLatestEntry(SyndFeed feed) {
+        List<SyndEntry> allEntries = feed.getEntries();
+        SyndEntry lastEntry;
+        if (allEntries.size() >= 1) {
+            /*
+             * The entries are stored in the SyndFeed object in the following order -
+             * the newest entry has index 0. The order is determined from the time the entry was posted, not the
+             * published time of the entry.
+             */
+            lastEntry = allEntries.get(0);
+
+            String title = lastEntry.getTitle();
+            String description = lastEntry.getDescription().getValue();
+            Date publishedDate = lastEntry.getPublishedDate();
+            String publishedDateString = DateFormat.getInstance().format(publishedDate);
+
+            String entryAsString = String.format("Title: %s%nDate: %s%nDescription: %s%n#", title, publishedDateString,
+                    description);
+            return entryAsString;
+        } else {
+            logger.debug("No entries found");
+            return new String();
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            SyndFeed feed = fetchFeedData(urlString);
```

Comment:
you might want to add some safeguard here as you might receive many REFRESH commands for the different channels in a row - it can be a problem if you refresh the feed on every call. So maybe check whether you have done an update in the past 5 seconds and only refresh, if it is already older than that?